cunumeric/module.py

```diff
@@ -1138,6 +1138,125 @@ def moveaxis(
 # Changing number of dimensions
 
 
+def _reshape_recur(ndim: int, arr: ndarray) -> tuple[int]:
+    if arr.ndim < ndim:
+        cur_shape = _reshape_recur(ndim - 1, arr)
+        if ndim == 2:
+            cur_shape = (1,) + cur_shape
+        else:
+            cur_shape = cur_shape + (1,)
+    else:
+        cur_shape = arr.shape
+    return cur_shape
+
+
+def _atleast_nd(
+    ndim: int, arys: Sequence[ndarray], view: Optional[bool] = True
```

Comment:
I think the proper type here is `view: bool`, since you only pass booleans, and the default value is also a boolean. `Optional[bool]` suggests that you can also pass `None` (or the default value can be `None`), which I don't think is your intention here. @bryevdv do I have it right?